src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+
+
+tableCellXWiki :: PandocMonad m => Bool -> [Block] -> XWikiReader m Text
+tableCellXWiki isHeader cell = do
+  contents <- blockListToXWiki cell
+  let cellBorder = if isHeader then "|=" else "|"
+  return $ cellBorder <> contents
+  
+
+inlineListToXWiki :: PandocMonad m => [Inline] -> XWikiReader m Text
+inlineListToXWiki lst =
+  fmap Data.Text.concat $ mapM inlineToXWiki lst
+
+
+inlineToXWiki :: PandocMonad m => Inline -> XWikiReader m Text
+
+inlineToXWiki (Str str) = return $ escapeXWikiString $ pack str
+
+inlineToXWiki Space = return " "
+
+inlineToXWiki LineBreak = return "\n"
+
+-- FIXME: better understand this
+inlineToXWiki SoftBreak = return " "
+
+inlineToXWiki (Emph lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "//" <> contents <> "//"
+
+inlineToXWiki (Strong lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "**" <> contents <> "**"
+
+inlineToXWiki (Strikeout lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "--" <> contents <> "--"
+
+inlineToXWiki (Superscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "^^" <> contents <> "^^"
+
+inlineToXWiki (Subscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ ",," <> contents <> ",,"
+
+-- TODO: Not supported. Maybe escape to HTML?
+inlineToXWiki (SmallCaps lst) = do
+  contents <- inlineListToXWiki lst
+  return contents
+
+inlineToXWiki (Quoted SingleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "'" <> contents <> "'"
+
+inlineToXWiki (Quoted DoubleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "\"" <> contents <> "\""
+
+inlineToXWiki (Code (_,classes,_) contents') = do
+  let at  = Set.fromList classes `Set.intersection` highlightingLangs
+  let contents = pack contents'
+  return $
+    case Set.toList at of
+      [] -> "{{code}}" <> contents <> "{{/code}}"
+      (l:_) -> "{{code language=\"" <> (pack l) <> "\"}}" <> contents <> "{{/code}}"
+
+inlineToXWiki (Cite _ lst) = inlineListToXWiki lst
+
+-- FIXME: optionally support this (plugin?) 
+inlineToXWiki (Math _ str) = return $ "{{formula}}" <> (pack str) <> "{{/formula}}"
+
+inlineToXWiki il@(RawInline frmt str)
+  | frmt == Format "xwiki" = return $ pack str
+  | otherwise              = "" <$ report (InlineNotRendered il)
+
+-- TODO: Handle anchors
+inlineToXWiki (Link _ txt (src, _)) = do
+  label <- inlineListToXWiki txt
+  case txt of
+     [Str s] | isURI src && escapeURI s == src -> return $ pack src
+     _  -> return $ "[[" <> label <> ">>" <> (pack src) <> "]]"
+
+inlineToXWiki (Image _ alt (source, tit)) = do
+  alt' <- inlineListToXWiki alt
+  let
+    titText = pack tit
+    params = intercalate " " [
+        if Text.null alt' then "" else "alt=\"" <> alt' <> "\"",
+          if Text.null titText then "" else "title=\"" <> titText <> "\""
+        ]
+  return $ "[[image:" <> (pack source) <> (if Text.null params then "" else "|| " <> params) <> "]]"
+
+inlineToXWiki (Note contents) = blockListToXWiki contents
```

Comment:
Is there a more graceful fallback?  (Maybe just printing `[`...`]` around the note's contents would help identify it as a note.)